src/super_gradients/training/models/model_factory.py

```diff
@@ -20,98 +22,108 @@
 logger = get_logger(__name__)
 
 
-def instantiate_model(name: str, arch_params: dict, pretrained_weights: str = None) -> SgModule:
+def get_architecture(model_name: str, arch_params: HpmStruct, pretrained_weights: str) -> Tuple[Type[torch.nn.Module], HpmStruct, str, bool]:
+    """
+    Get the corresponding architecture class.
+
+    :param model_name:          Define the model's architecture from models/ALL_ARCHITECTURES
+    :param arch_params:         Architecture hyper parameters. e.g.: block, num_blocks, etc.
+    :param pretrained_weights:  Describe the dataset of the pretrained weights (for example "imagenent")
+
+    :return:
+        - architecture_cls:     Class of the model
+        - arch_params:          Might be updated if loading from remote deci lab
+        - pretrained_weights:   Might be updated if loading from remote deci lab
+        - is_remote:            True if loading from remote deci lab
+    """
+    is_remote = False
+    if not isinstance(model_name, str):
+        raise ValueError("Unsupported model model_name " + str(model_name) + ", see docs or all_architectures.py for supported nets.")
+    elif isinstance(model_name, str) and model_name not in ARCHITECTURES.keys():
```

Comment:
no need to check isinstance(model_name, str) here (otherwise it would have go into the if)

Reply:
Updated in dd505a03c6cd553696f7d3e80ef8dfc25992af4c